www/class/centreon-clapi/centreonHost.class.php

```diff
@@ -334,6 +335,97 @@ public function setinstance($parameters)
     }
 
     /**
+     * Get a parameter
+     *
+     * @param string $parameters
+     * @return void
+     * @throws CentreonClapiException
+     */
+    public function getparam($parameters = null)
+    {
+        $params = explode($this->delim, $parameters);
+        if (count($params) < 2) {
+            throw new CentreonClapiException(self::MISSINGPARAMETER);
+        }
+        if (($objectId = $this->getObjectId($params[self::ORDER_UNIQUENAME])) != 0) {
+            $extended = false;
+            $field = $params[1];
+            switch ($params[1]) {
+                case "check_command":
+                    $field = "command_command_id";
+                    break;
+                case "check_command_arguments":
+                    $field = "command_command_id_arg1";
+                    break;
+                case "event_handler":
+                    $field = "command_command_id2";
+                    break;
+                case "event_handler_arguments":
+                    $field = "command_command_id_arg2";
+                    break;
+                case "check_period":
+                    $field = "timeperiod_tp_id";
+                    break;
+                case "notification_period":
+                    $field = "timeperiod_tp_id2";
+                    break;
+                case "notes":
+                case "notes_url":
+                case "action_url":
+                case "icon_image":
+                case "icon_image_alt":
+                case "vrml_image":
+                case "statusmap_image":
+                case "2d_coords":
+                case "3d_coords":
+                    $extended = true;
+                    break;
+                case self::HOST_LOCATION:
+                    $field = 'host_location';
+                    break;
+                default:
+                    if (!preg_match("/^host_/", $params[1])) {
+                        $field = "host_" . $params[1];
+                    }
+                    break;
+            }
+            if ($extended == false) {
+                $ret = $this->object->getParameters($objectId, $field);
+                $ret = $ret[$field];
+            } else {
+                $field = "ehi_" . $field;
+                $extended = new \Centreon_Object_Host_Extended();
+                $ret = $extended->getParameters($objectId, $field);
+                $ret = $ret[$field];
+            }
+            switch ($params[1]) {
```

Comment:
Is this the same as the previous switch

Reply:
Yes. But this one is to convert parameter value if needed. Otherwise, the ->getParameters would be duplicated.